### PR TITLE
Enable IPv6 support on CC32xx

### PR DESCRIPTION
**Pull request by vanti**
IPv6 support was not fully implemented in the driver and was disabled.
This commit completes the implementation and enables it when
configured.

This is validated using the echo_async sample with a minor modification.
On TI CC32xx, the same port number cannot be bound to different
sockets. Instead, when an IPv6 socket is bound it can handle both IPv4
and IPv6 connections. So for this platform we only create the IPv6 socket.

### CODE REVIEW

**Review by tbursztyka — APPROVED** [review]

File: drivers/wifi/simplelink/simplelink.c [hunk 1]
```diff
@@ -24,6 +24,9 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #define SCAN_RETRY_DELAY 2000  /* ms */
 #define FC_TIMEOUT K_SECONDS(CONFIG_WIFI_SIMPLELINK_FAST_CONNECT_TIMEOUT)
 
+#define IPV4 0x1
```
Comment (chrta) [inline drivers/wifi/simplelink/simplelink.c:27]: I think there is a chance that defines with the same name are used somewhere else. I would prefer if these defines here would be prefixed, e.g. `IPV4` -> `SIMPLELINK_IPV4`

**Review by chrta — CHANGES_REQUESTED** [review]

File: drivers/wifi/simplelink/simplelink.c [hunk 2]
```diff
@@ -202,6 +217,14 @@ static void simplelink_iface_init(struct net_if *iface)
...
 
 	simplelink_data.iface = iface;
+	simplelink_data.mask = 0;
+
+#ifdef CONFIG_NET_IPV4
+	simplelink_data.mask |= IPV4;
+#endif
+#ifdef CONFIG_NET_IPV6
+	simplelink_data.mask |= IPV6;
+#endif
```
Comment (jukkar) [inline drivers/wifi/simplelink/simplelink.c:227]: You could simplify this by doing it like this:
```
simplelink_data.mask |= IS_ENABLED(CONFIG_NET_IPV4) ? SIMPLELINK_IPV4 : 0;
simplelink_data.mask |= IS_ENABLED(CONFIG_NET_IPV6) ? SIMPLELINK_IPV6 : 0;
```

**Review by chrta — APPROVED** [review]
LGTM

**Comment by vanti** [comment]
Thank you all for the reviews!

**Pull request merged**